src/Microsoft.AspNet.Mvc.Core/DefaultControllerActionArgumentBinder.cs

```diff
@@ -57,11 +59,82 @@ public class DefaultControllerActionArgumentBinder : IControllerActionArgumentBi
                 }
             }
 
+            var controllerType = actionDescriptor.ControllerTypeInfo.AsType();
+            var propertyMetadata = new List<ModelMetadata>();
+            foreach (var property in actionDescriptor.CommonParameters)
+            {
+                var metadata = _modelMetadataProvider.GetMetadataForProperty(
+                    controllerType,
+                    propertyName: property.Name);
```

Comment:
BinderName is already part of the MBC, which keeps adding up as we traverse, the BindingSource does not make sense in MBC because it differs for every property/parameter while binding. 
What you are saying for removing ```GetModelMetadataForParameter ``` makes sense, we can just remove it and use the type metadata and the descriptor as you suggested.